### ISSUE TITLE
[Future] Rules

### ISSUE THREAD

**Issue by derekantrican** [derekantrican/GAS-ICS-Sync#69]
*This is a future feature that probably won't be implemented for a while. We have bigger things to focus on before then*

**Filter Rules:** only process events that match certain criteria

*Examples:*

- "Do not process any event where the title contains 'test'"
- "Do not process any events on Fridays"
- "Do not process any events occurring past 11/1/19"

**Modification rules:** modify event properties before processing the event

*Examples:*

- "Remove 'FW:' from any event titles" (as requested in #68)
- "Replace Title with {Description} and Description with {Title}" (would be a better approach than our current "flag" setting)
- "Modify all event start times to happen at 23:00 and end times to happen at 23:59" (as requested in #66)

-------------

Rules should be able to access the following properties: Title, Location, Description, Start, End, Duration

**Comment by derekantrican** [comment]
Some people may also desire in the future the ability to automatically modify the colors of the events. This could also be done via rules. For instance:

- "All event colors should be red" (meaning all synced/ICS events - not all Google calendar events)
- "All events where the title contains 'Doctor's appointment' should be blue"

**Comment by jstride** [comment]
The modification use cases could potentially be split into two features, rules and regexs. 

The first modifications (Remove FW:, and others such as replace X with Y, or remove that annoying 'Invitation/Updated' Google adds to events sent to other calendars) could be solved by having a config option as a regex to run for title and description, as a thought.

**Comment by derekantrican** [comment]
Yeah, these will probably support regex (or at least have an option to make the modification rule "regex" or "non regex")
